R/group_apply.R

```diff
@@ -0,0 +1,119 @@
+#' Apply a function over grouping(s) of data
+#'
+#' A function to apply `palaeoverse` functionality across subsets (groups) of
+#' data, delineated using one or more variables. `data.frame` related functions
+#' (e.g. `nrow`, `ncol`, `lengths`, `unique`) may also be used.
+#'
+#' @param occdf \code{dataframe}. A dataframe of fossil occurrences.
```

Comment:
not sure about calling it "occdf" - for `tax_check` it would rather be a taxdf. This can result in messages like

```
group_apply(occdf = occdf[,1:3],
group = c("collection_no"),
fun = tax_check,verbose = FALSE)
```

Error in FUN(X[[i]], ...) : 
  Please specify `name` as a single column name in `taxdf`

Reply:
I still wonder whether we could use "occdf" as the name for the df argument for taxdf

Reply:
Hi Kilian, I think you got this error because you didn't update the `name` argument in tax_check. The default is the "genus" column. We could update taxdf to occdf in tax_check. It works with both. I think taxdf just makes it more clear to the user that it should contain taxonomic info?

Reply:
I've updated taxdf to occdf in group_apply :-)